src/components/BorrowerProfile/RepaymentSchedule.vue

```diff
@@ -0,0 +1,233 @@
+				flexibleFundraisingEnabled
+				lenderRepaymentTerm
+				expectedPayments {
+					amount
+					localAmount
+					dueToKivaDate
+					effectiveDate
+				}
+			}
+			... on LoanPartner {
+				partner {
+					id
+					name
+				}
+			}
+		}
+	}
+}`;
+
+export default {
+	components: {
+		KvLightbox,
+	},
+	inject: ['apollo'],
+	props: {
+		loanId: {
+			type: Number,
+			default: 0,
+		},
+		status: {
+			type: String,
+			default: '',
+		},
+	},
+	data() {
+		return {
+			// mdiCheckboxMarkedCircle,
+			// mdiMinusCircle,
+			isLightboxVisible: false,
+			firstRepaymentDate: '',
+			repaymentSchedule: [],
+			repaidAmount: 0,
+			loanAmount: 0,
+			lenderRepaymentTerm: 0,
+			partnerName: '',
+		};
+	},
+	methods: {
+		openLightbox() {
+			this.isLightboxVisible = true;
+		},
+		closeLightbox() {
+			this.isLightboxVisible = false;
+		},
+		calculateRepaymentSchedule() {
+			this.apollo.query({
+				query: repaymentScheduleQuery,
+				variables: {
+					loanId: this.loanId
+				}
+			}).then(({ data }) => {
+				this.partnerName = data?.lend?.loan?.partner?.name || '';
+				this.repaymentSchedule = data?.lend?.loan?.terms?.expectedPayments || [];
+				this.repaidAmount = data?.lend?.loan?.paidAmount || 0;
+				this.loanAmount = data?.lend?.loan?.loanAmount || 0;
+				this.lenderRepaymentTerm = data?.lend?.loan?.terms?.lenderRepaymentTerm || 0;
+				if (this.isPartnerLoan) {
+					this.firstRepaymentDate = this.repaymentSchedule[0].dueToKivaDate || '';
+				}
+			});
+		},
+	},
+	computed: {
+		isPartnerLoan() {
+			return !!this.partnerName;
+		},
+		formattedFirstRepaymentDate() {
+			if (this.firstRepaymentDate !== '') {
+				return format(parseISO(this.firstRepaymentDate), 'MMM yyyy');
+			}
+			return false;
+		},
+		statusLanguageCheck() {
+			if (this.status === 'fundraising') {
+				return 'begin';
+			}
+			return 'began';
+		},
+		repaymentStatusCheck() {
+			if (this.status === 'fundraising') {
+				return 'on track';
+			}
+			// TODO: fill out other options for other loan statuses
+			return 'on track';
+		},
+		formatRepaymentSchedule() {
+			const formattedRepaymentSchedule = [];
+			if (this.repaymentSchedule !== []) {
+				this.repaymentSchedule.forEach(repayment => {
+					formattedRepaymentSchedule.push({
+						repaymentDateFormatted: format(parseISO(repayment.dueToKivaDate), 'MMM yyyy'),
+						repaymentAmountFormatted: numeral(repayment.amount).format('$0,0.00'),
+						// Dupe of above line
+						repaymentDateExpectedFormatted: format(parseISO(repayment.dueToKivaDate), 'MMM yyyy'),
+					});
+				});
+			}
+			return formattedRepaymentSchedule;
+		},
+		loanAmountFormatted() {
+			return numeral(this.loanAmount).format('$0,0.00');
+		},
+		calculateMonthlyPayment() {
+			// Used for calculating the monthly payment of a direct loan
+			return numeral(this.loanAmount / this.lenderRepaymentTerm).format('$0,0.00');
+		}
+	},
+	mounted() {
+		this.calculateRepaymentSchedule();
+	},
+};
+</script>
+
+<style lang="scss" scoped>
```

Comment:
This is no problem, and a good use case for using Tailwind's @ apply postcss stuff. Ideally we're never hard-coding values like 1.25rem anymore, since we have abstracted spacing to our design tokens / Tailwind.

You can use the tailwind in a Vue `style` tag using `@apply` and `lang="postcss"`. Some more details in this PR: https://github.com/kiva/ui/pull/3362

For this it would look something like
```
<style lang="postcss">
.table-heading-spacing {
  @apply tw-py-2.5 tw-pl-1.5;
}

.table-heading-spacing {
  @apply tw-p-1.5;
}
</style>

```

Reply:
Should the `scoped` tag remain too?

Reply:
Yes please!

Reply:
Thanks for this advice Ryan!. I've just committed this update.